cassandra-4/src/main/java/io/debezium/connector/cassandra/AbstractCassandra4CommitLogParser.java

```diff
@@ -0,0 +1,116 @@
+    protected int pollingInterval;
+
+    public AbstractCassandra4CommitLogParser(Cassandra4CommitLogProcessor.LogicalCommitLog commitLog, final List<ChangeEventQueue<Event>> queues,
+                                             final CommitLogProcessorMetrics metrics,
+                                             final CassandraConnectorContext context) {
+        this.commitLogReader = new CommitLogReader();
+        this.queues = queues;
+        this.metrics = metrics;
+        this.commitLog = commitLog;
+
+        this.commitLogReadHandler = new Cassandra4CommitLogReadHandlerImpl(
+                context.getSchemaHolder(),
+                context.getQueues(),
+                context.getOffsetWriter(),
+                new RecordMaker(context.getCassandraConnectorConfig().tombstonesOnDelete(),
+                        new Filters(context.getCassandraConnectorConfig().fieldExcludeList()),
+                        context.getCassandraConnectorConfig()),
+                metrics);
+
+        this.commitLogTransfer = context.getCassandraConnectorConfig().getCommitLogTransfer();
+        this.erroneousCommitLogs = context.getErroneousCommitLogs();
+        this.pollingInterval = context.getCassandraConnectorConfig().getCommitLogMarkedCompletePollInterval();
+    }
+
+    public void complete() {
+        completePrematurely = true;
+    }
+
+    public abstract Cassandra4CommitLogProcessor.ProcessingResult parse();
+
+    public Cassandra4CommitLogProcessor.ProcessingResult process() {
+        if (!commitLog.exists()) {
+            LOGGER.warn("Commit log " + commitLog + " does not exist!");
+            return new Cassandra4CommitLogProcessor.ProcessingResult(commitLog, Cassandra4CommitLogProcessor.ProcessingResult.Result.DOES_NOT_EXIST);
+        }
+
+        LOGGER.info("Processing commit log {}", commitLog.log.toString());
+        metrics.setCommitLogFilename(commitLog.log.toString());
+        metrics.setCommitLogPosition(0);
+
+        Cassandra4CommitLogProcessor.ProcessingResult result = parse();
+        enqueueEOFEvent();
+        Cassandra4CommitLogProcessor.removeProcessing(this);
+        LOGGER.debug("Processing {} callables.", Cassandra4CommitLogProcessor.submittedProcessings.size());
+        return result;
+    }
+
+    protected void enqueueEOFEvent() {
+        try {
+            queues.get(Math.abs(commitLog.log.getName().hashCode() % queues.size())).enqueue(new EOFEvent(commitLog.log));
+        }
+        catch (InterruptedException e) {
+            throw new CassandraConnectorTaskException(String.format(
+                    "Enqueuing has been interrupted while enqueuing EOF Event for file %s", commitLog.log.getName()), e);
+        }
+    }
+
+    protected void processCommitLog(Cassandra4CommitLogProcessor.LogicalCommitLog logicalCommitLog, CommitLogPosition position) {
+        try {
+            LOGGER.debug("starting to read commit log segments {} on position {}", logicalCommitLog, position);
```

Comment:
Could you upper-case this? I am sorry I left it like that (I think it was me). Same with the next debug log below.